processors/data_contributions/processor.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import requests
+import logging
+from .. import base
+from . import extractors as extractors_module
+logger = logging.getLogger(__name__)
+
+
+def process(conf, conn):
+    def convert_ids(contrib):
+        converted_ids = {key: val.hex for key, val in contrib.items()
+                        if val.__class__.__name__ == 'UUID'}
+        contrib.update(converted_ids)
+        return contrib
+
+    extractors = base.helpers.get_variables(
+        extractors_module, lambda x: x.startswith('extract_'))
+
+    # Upsert source
+    source = extractors['extract_source']()
+    source_id = base.writers.write_source(conn, source)
+
+    # Upsert documents from data contributions
+    for_processing = """SELECT * FROM data_contributions
+                        WHERE approved=true
+                        AND trial_id IS NOT NULL
+                        AND document_category_id IS NOT NULL
+                        AND (data_url IS NOT NULL OR url IS NOT NULL);"""
+    for data_contrib in conn['explorer'].query(for_processing):
+        data_contrib = convert_ids(data_contrib)
+        trial = conn['database']['trials'].find_one(id=data_contrib['trial_id'])
+        if not trial:
+            continue
+        _process_document(conn, data_contrib, extractors, source_id)
+
+    # Remove documents from data_contributions
+    for_removal = """SELECT * FROM data_contributions
+                     WHERE approved=false
+                     AND document_id IS NOT NULL;"""
+    for data_contrib in conn['explorer'].query(for_removal):
+        data_contrib = convert_ids(data_contrib)
+        _remove_document(conn, data_contrib)
+
+
+def _process_document(conn, contrib, extractors, source_id):
+    document_id = None
+    document = extractors['extract_document'](contrib)
+    data_request = _request_data_contribution(document['source_url'])
+    if data_request:
+        content_type = data_request.headers['Content-Type']
```

Comment:
It feels overkill to download a potentially large document just to check its content type. It would be better to either validate the URL's file extension (as in `.pdf`, `.zip`, etc) or at least use a HEAD request instead of a GET, so we don't actually download the file. I think it's fine to check the file extension only.

Readability could be improved with better naming and extracting a method. For example, you could do something like:

```python
def _process_document(conn, contrib, extractors, source_id):
    if not _document_has_valid_format(document):
        return
    # Process the document here


def _document_has_valid_format(document):
    # Check if the format is valid
```

I think it would also be better to pass the document itself to the `_process_document()` function, instead of sending in the `extractors`. This way all extracting happens on the `process()` function. The method signature would become `_process_document(conn, document, data_contribution, source_id)`